fluvio/new_topic.py

```diff
@@ -0,0 +1,124 @@
+from dataclasses import dataclass, replace
+from typing import Optional, List, Union
+from enum import Enum
+from .specs import TopicSpec, PartitionMap
+from .utils import parse_byte_size
+from humanfriendly import parse_timespan
+
+
+class TopicMode(Enum):
```

Comment:
Instead of creating python enum, should use derive macro like here: https://pyo3.rs/v0.23.2/conversions/traits.html

Reply:
This new enum is only for the builder of TopicSpec, if we consider the solution without the build it's not even needed.

Reply:
Maybe the Compression Enum is worth, thought